### PR TITLE
fix(typos): fixed a variety of typos throughout project

### PR DESCRIPTION
**Pull request by mwtoews**
This is a bit more comprehensive than #82 using [codespell](https://github.com/codespell-project/codespell).

Most changes are comments, some are documentation, and other changes may have other impacts.

### CODE REVIEW

File: doc/ReleaseNotes/previous/v6.4.2.tex [hunk 1]
```diff
@@ -10,7 +10,7 @@
...
...her retrieve all input data from named locations, called memory paths, allocated in managed memory.  The collection of all simulation input data in managed memory is called the input context.  IDP uses existing descriptions of input variables, called variable definitions, to interpret and store input.  The program variable definition set and its representation in the input context is described as the Input Data Model (IDM).  Input variables can be recognized in a memory dump (e.g., with the MEMORY\_PRINT\_OPTION activated in the options block of mfsim.nam) by their memory path prefix string "\_\_INPUT\_\_".  Components that later access input typically copy data from the input context to their own memory space; therefore, the present implementation of the IDP results in an increased memory footprint for the program.  Among its advantages include the consolidation of all input processing early in program runtime, and outside of any particular component, enabling the support of alternative types of input data sources.  Input file types that are currently processed by IDP include DIS6, DISU6, DIV6, NPF6, DSP6, and Name File inputs for the Simulation (mfsim.nam) and GWF and GWT models.
```
Comment (mwtoews) [inline doc/ReleaseNotes/previous/v6.4.2.tex:13]: For some reason, the GH UI does not highlight the one change here "implemention" -> "implementation"

File: src/Model/GroundWaterFlow/gwf-lak.f90 [hunk 2]
```diff
@@ -3091,7 +3091,7 @@ subroutine lak_set_stressperiod(this, itemno)
       bndElem => this%withdrawal(itemno)
       call read_value_or_time_series_adv(text, itemno, jj, bndElem, &
                                          this%packName, 'BND', this%tsManager, &
-                                         this%iprpak, 'WITHDRAWL')
+                                         this%iprpak, 'WITHDRAWAL')
```
Comment (mwtoews) [inline src/Model/GroundWaterFlow/gwf-lak.f90:3094]: This typo is possibly the source of a bug (reading the context of the surrounding code)

**Comment by mwtoews** [comment]
Note that currently changes applied to .dfn files are not applied to .md or .tex formats.

Also [CONTRIBUTING.md](https://github.com/MODFLOW-USGS/modflow6/blob/develop/CONTRIBUTING.md) or other docs don't say if PR changes to .dfn files should be propagated to derivative formats or not, or if that happens later. Clarification would be helpful.

**Comment by wpbonelli** [comment]
IMO derivative files should generally stay in sync with dfns, `doc/mf6io/mf6ivar/mf6ivar.py` does this. It's ok if this doesn't happen right away as it can always be corrected later (updates run before building docs or distributions) but we could add a PR check. I'll add a section to the dev docs too.

See for instance https://github.com/MODFLOW-USGS/modflow6/commit/b8bdf6106fdae5c290389fdfbbac30c117b03b67 where dfns, `mf6ivar.md`, and `.dat`/`.tex` files are updated together

**Review by wpbonelli — APPROVED** [review]

**Pull request merged**